### PR TITLE
adjusting documentation typesetting

### PR DESCRIPTION
**Pull request by rjain37**
------------------------------------------------------------------------------------------------------------

**Context:** Documentation on some quantum gates are oddly formatted.

**Description of the Change:** The identity gate uses $\I$ instead of $I$ where $\I$ seems to be defined as $1^\hat{}$ or something along those lines. Otherwise, a lot of functions that use $\exp$ don't use \left( and \right) when typesetting delimiters containing fractions. 

**Benefits:** "Better" typesetting.

**Possible Drawbacks:** None(?)

**Related GitHub Issues:** n/a

### CODE REVIEW

**Review by albi3ro — APPROVED** [review]
Thanks for opening this Pull Request @rjain37 . These changes definitely make things look cleaner.

You can now add your name to the list of contributors at the bottom of our changelog, and add a changelog entry under the [`Documentation`](https://github.com/PennyLaneAI/pennylane/blob/007fef3788e5f98e5dfe1fbdf594a2143bb9d6fb/doc/releases/changelog-dev.md?plain=1#L219) section.

https://github.com/PennyLaneAI/pennylane/blob/master/doc/releases/changelog-dev.md

**Comment by albi3ro** [comment]
Note that codecov can sometime act up for no reason, so we can override codecov if it continues to complain.

File: doc/releases/changelog-dev.md [hunk 1]
```diff
@@ -216,6 +216,8 @@
 * A small typo was fixed in the docstring for `qml.sample`.
   [(#5685)](https://github.com/PennyLaneAI/pennylane/pull/5685)
 
+* Typesetting for some of the documentation was fixed, (use of left/right delimiters, fractions, and fix of incorrectly set up commands) [(#5804)](https://github.com/PennyLaneAI/pennylane/pull/5804)
```
Comment (mudit2812) [inline doc/releases/changelog-dev.md:219]:
```suggestion
* Typesetting for some of the documentation was fixed, (use of left/right delimiters, fractions, and fix of incorrectly set up commands)
  [(#5804)](https://github.com/PennyLaneAI/pennylane/pull/5804)
```

File: doc/releases/changelog-dev.md [hunk 2]
```diff
@@ -301,4 +303,5 @@ Lee James O'Riordan,
 Mudit Pandey,
 Kenya Sakka,
 Haochen Paul Wang,
-David Wierichs.
+David Wierichs,
+Rohan Jain.
```
Comment (mudit2812) [inline doc/releases/changelog-dev.md:307]: The contributors names should be sorted alphabetically by last name.

**Review by mudit2812 — APPROVED** [review]
Looks good. I'll merge once the changelog formatting is fixed. Thanks for spotting and fixing this @rjain37

**Comment by rjain37** [comment]
@mudit2812 thanks for pointing that out. should be good to go now!

**Pull request merged**